lib/token.js

```diff
@@ -179,10 +179,13 @@ function handleOAuthResponse(sdk, oauthParams, res, urls) {
       var validationParams = {
         clientId: clientId,
         issuer: urls.issuer,
-        nonce: oauthParams.nonce,
-        ignoreSignature: oauthParams.ignoreSignature
+        nonce: oauthParams.nonce
       };
 
+      if (oauthParams.ignoreSignature !== undefined) {
+        validationParams.ignoreSignature = oauthParams.ignoreSignature;
```

Comment:
So if oauthParams.ignoreSignature IS `undefined`, we'll not set it to `undefined` and thus leave it as...`undefined`?

Is the `if` needed?

Reply:
This is working around some subtle behavior in the `util.extend` method used in the `verifyToken ()` method: https://github.com/okta/okta-auth-js/pull/205/files/7650c24813f25bf019b21bab3a1e09fd5dd1e324#diff-398ed3b014436a5204583323ea29320bR57

The tricky behavior is that if the key in the object is defined, it will be used (even if the value is undefined). So this check is to protect the default value used in the OAuth constructor from being overridden with 'undefined'.

I would guess this behavior can produce unexpected results in other places, perhaps `util.extend` should be fixed to not merge keys with an 'undefined' value.

Reply:
Explanation makes sense - add a comment to explain why?